spec/models/validations/sales/setup_validations_spec.rb

```diff
@@ -36,19 +31,27 @@
         end
       end
 
-      context "when saledate is before the 22/23 collection year" do
+      context "when saledate is before the open collection year" do
         let(:record) { build(:sales_log, saledate: Time.zone.local(2020, 1, 1)) }
 
+        before do
+          allow(Time).to receive(:now).and_return(Time.zone.local(2023, 1, 10))
+        end
+
         it "adds error" do
           setup_validator.validate_saledate_collection_year(record)
 
           expect(record.errors[:saledate]).to include("Enter a date within the 22/23 collection year, which is between 1st April 2022 and 31st March 2023")
         end
       end
 
-      context "when saledate is after the 22/23 collection year" do
+      context "when saledate is after the open collection year" do
```

Comment:
similar to the one above, if you're happy testing the general format with regex you could dodge all the time based tomfoolery with saledate: today + n.years